core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -82,6 +85,13 @@ public UnsafeSorterSpillReader(
       Closeables.close(bs, /* swallowIOException = */ true);
       throw e;
     }
+    taskContext.addTaskCompletionListener(context -> {
+      try {
+        close();
+      } catch (IOException e) {
```

Comment:
For my understanding, what happens if we fail to close here? `Thread Leak`s still happen?

Reply:
I don't think there should be a leak, executionService.shutdownNow() will always be called.